### PR TITLE
New rules and queries for Cisco Umbrella, Cisco ISE and Proofpoint POD

### PR DESCRIPTION
**Pull request by vu-socprime**
Added new detection rules and hunting queries for Cisco Umbrella, Cisco ISE and Proofpoint POD

### CODE REVIEW

**Comment by vu-socprime** [comment]
During KqlValidation there is an error "The name 'ProofpointPOD' does not refer to any known table, tabular variable or function". But all queries are based on functions (ProofpointPOD is the name of function) located in "Parsers" directory. Could you please tell if it is possible to pass KqlValidation in this situation?

**Comment by shainw** [comment]
> During KqlValidation there is an error "The name 'ProofpointPOD' does not refer to any known table, tabular variable or function". But all queries are based on functions (ProofpointPOD is the name of function) located in "Parsers" directory. Could you please tell if it is possible to pass KqlValidation in this situation?

@vu-socprime - You will need to create a CustomTable mapping for these as these connectors are not built into the build system at this point.  Instructions are here - https://github.com/Azure/Azure-Sentinel#pull-request-kql-validation-check - specifically this -

`If you are using custom logs table (a table which is not defined on all workspaces by default) you should verify your table schema is defined in json file in the folder Azure-Sentinel\.script\tests\KqlvalidationsTests\CustomTables`

**Comment by vu-socprime** [comment]
> > During KqlValidation there is an error "The name 'ProofpointPOD' does not refer to any known table, tabular variable or function". But all queries are based on functions (ProofpointPOD is the name of function) located in "Parsers" directory. Could you please tell if it is possible to pass KqlValidation in this situation?
> 
> @vu-socprime - You will need to create a CustomTable mapping for these as these connectors are not built into the build system at this point. Instructions are here - https://github.com/Azure/Azure-Sentinel#pull-request-kql-validation-check - specifically this -
> 
> `If you are using custom logs table (a table which is not defined on all workspaces by default) you should verify your table schema is defined in json file in the folder Azure-Sentinel\.script\tests\KqlvalidationsTests\CustomTables`

@shainw, json files for custom tables are already in Azure-Sentinel\.script\tests\KqlvalidationsTests\CustomTables. The issue is that we use functions (Azure-Sentinel/Parsers/) instead of tables because we need normalized fields to make queries. What should we do in this situation?

**Comment by shainw** [comment]
@vu-socprime - You can include it in the CustomTable, but it would be a JSON with the schema of the Parser versus the Table.  At least I believe this should work.  Give it a try and see if it allows passing of the validation.

**Comment by vu-socprime** [comment]
@shainw - Ok, I will try it. Thank you.

**Comment by shainw** [comment]
> @shainw - Ok, I will try it. Thank you.

@vu-socprime - I think you did the wrong one.  I see you submitted for Cisco_Umbrella but the KQL validation error is on the ProofPointPOD missing a customtable entry.

**Comment by vu-socprime** [comment]
> > @shainw - Ok, I will try it. Thank you.
> 
> @vu-socprime - I think you did the wrong one. I see you submitted for Cisco_Umbrella but the KQL validation error is on the ProofPointPOD missing a customtable entry.

@shainw - There were errors regarding Cisco_Umbrella, ProofPointPOD and CiscoISEEvent. I've also added schemas for  ProofPointPOD and CiscoISEEvent in last commit.

File: Detections/ProofpointPOD/ProofpointPODHighRiskNotDiscarded.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - InitialAccess
+query: |
+  let lbtime = 10m;
+  ProofpointPOD
+  | where TimeGenerated > ago(lbtime)
+  | where EventType == 'message'
+  | where NetworkDirection == 'inbound'
+  | where FilterDisposition !in ('reject', 'discard')
+  | where FilterModulesSpamScoresOverall == '100'
+  | project SrcUserUpn, DstUserUpn
```
Comment (shainw) [inline Detections/ProofpointPOD/ProofpointPODHighRiskNotDiscarded.yaml:24]: Missing the Entity Mapping section on this one.

File: Detections/ProofpointPOD/ProofpointPODWeakCiphers.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+query: |
+  let lbtime = 1h;
+  let tls_ciphers = dynamic(['RC4-SHA', '​DES-CBC3-SHA']);
+  ProofpointPOD
+  | where EventType == 'message'
+  | where TlsCipher in (tls_ciphers)
```
Comment (shainw) [inline Detections/ProofpointPOD/ProofpointPODWeakCiphers.yaml:19]: Missing the Entity Mapping section

**Review by shainw — CHANGES_REQUESTED** [review]
Looks like referencing the parser in the CustomTable worked.  Additionally, a couple of required changes for Detections with the Entity Mapping.  Also, to support a future improvement we want to include Entity Mapping on Hunting queries also now.  This is not required, but when it does become required we would need you to come back and do it then, so may want to do it now.

**Review by shainw — APPROVED** [review]

**Pull request merged**